src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private double width;
+    private double height;
+
+    public Rectangle(Color color, double width, double height) {
+        super(color);
+        this.width = width;
+        this.height = height;
+    }
+
+    @Override
+    public double getArea() {
+        return width * height;
+    }
+
+    @Override
+    public void print() {
+        System.out.println("Figure: rectangle, area: " + getArea() + " sq. units, width: "
+                + width + " units, height: " + height + " units, color: "
+                + new ColorSupplier().getRandomColor());

```

Comment:
Avoid creating a new instance of `ColorSupplier` in the `print()` method. This can lead to less control over the color generation process and can make testing more difficult. Consider passing a `ColorSupplier` instance to the class through the constructor and reusing it.